### PR TITLE
refactor: rename notices

### LINKED ISSUE #793: Improve notice name consistency
Revisit notices class names for consistency.

[...] I'd suggest we put the entity name first when it makes sense for consistency, and because if we list the rules in the table alphabetically then, for example, most of the Route rules should be grouped together. So, for example, this class and code could be renamed to `RouteSameNameAndDescriptionNotice` and `route_same_name_and_description`. Then it would appear right after `RouteBothShortAndLongNameMissingNotice`.

_Originally posted by @barbeau in https://github.com/MobilityData/gtfs-validator/pull/781#discussion_r585866237_

### PR DESCRIPTION
**Pull request by lionel-nj**
closes #793 

**Summary:**

This PR revisits some notice class names as suggested by https://github.com/MobilityData/gtfs-validator/pull/781#discussion_r585866237

> Also, I'd suggest we put the entity name first when it makes sense for consistency, and because if we list the rules in the table alphabetically then, for example, most of the Route rules should be grouped together. So, for example, this class and code could be renamed to RouteSameNameAndDescriptionNotice and route_same_name_and_description. Then it would appear right after RouteBothShortAndLongNameMissingNotice.

**Expected behavior:** 

No code logic change. Tests should pass.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- ~[ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

File: main/src/main/java/org/mobilitydata/gtfsvalidator/notice/StopLocationWithoutParentStationNotice.java [hunk 1]
```diff
@@ -26,8 +26,8 @@
  *
  * <p>Severity: {@code SeverityLevel.ERROR}
  */
-public class LocationWithoutParentStationNotice extends ValidationNotice {
-  public LocationWithoutParentStationNotice(
+public class StopLocationWithoutParentStationNotice extends ValidationNotice {
```
Comment (aababilov) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/notice/StopLocationWithoutParentStationNotice.java:29]: What is the rationale for that change? This notice applies to stops, entrances and boarding areas.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/notice/AgencyLangInconsistencyNotice.java [hunk 2]
```diff
@@ -23,8 +23,8 @@
  *
  * <p>Severity: {@code SeverityLevel.WARNING}
  */
-public class InconsistentAgencyLangNotice extends ValidationNotice {
-  public InconsistentAgencyLangNotice(long csvRowNumber, String expected, String actual) {
+public class AgencyLangInconsistencyNotice extends ValidationNotice {
```
Comment (aababilov) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/notice/AgencyLangInconsistencyNotice.java:26]: What is the rationale for this change?

File: main/src/main/java/org/mobilitydata/gtfsvalidator/notice/AgencyTimezoneInconsistencyNotice.java [hunk 3]
```diff
@@ -23,8 +23,8 @@
  *
  * <p>Severity: {@code SeverityLevel.ERROR}
  */
-public class InconsistentAgencyTimezoneNotice extends ValidationNotice {
-  public InconsistentAgencyTimezoneNotice(long csvRowNumber, String expected, String actual) {
+public class AgencyTimezoneInconsistencyNotice extends ValidationNotice {
```
Comment (aababilov) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/notice/AgencyTimezoneInconsistencyNotice.java:26]: What is the rationale for this change?

File: main/src/main/java/org/mobilitydata/gtfsvalidator/notice/RouteSameNameAndDescriptionNotice.java [hunk 4]
```diff
@@ -45,6 +45,6 @@ public SameNameAndDescriptionForRouteNotice(
 
   @Override
   public String getCode() {
-    return "same_route_name_and_description";
+    return "route_same_name_and_description_notice";
```
Comment (aababilov) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/notice/RouteSameNameAndDescriptionNotice.java:48]: The _notice suffix is redundant.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/notice/RouteSameNameAndDescriptionNotice.java [hunk 5]
```diff
@@ -29,8 +29,8 @@
  *
  * <p>Severity: {@code SeverityLevel.ERROR}
  */
-public class SameNameAndDescriptionForRouteNotice extends ValidationNotice {
-  public SameNameAndDescriptionForRouteNotice(
+public class RouteSameNameAndDescriptionNotice extends ValidationNotice {
```
Comment (aababilov) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/notice/RouteSameNameAndDescriptionNotice.java:32]: I feel that RouteWithSameNameAndDescriptionNotice sounds more natural.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/notice/StopPlatformWithoutParentStationNotice.java [hunk 6]
```diff
@@ -25,8 +25,8 @@
  *
  * <p>Severity: {@code SeverityLevel.WARNING}
  */
-public class PlatformWithoutParentStationNotice extends ValidationNotice {
-  public PlatformWithoutParentStationNotice(long csvRowNumber, String stopId, String stopName) {
+public class StopPlatformWithoutParentStationNotice extends ValidationNotice {
```
Comment (aababilov) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/notice/StopPlatformWithoutParentStationNotice.java:28]: What is the rationale for this change?

**Review by aababilov — CHANGES_REQUESTED** [review]
Which principle was applied for choosing new names for these classes? I see that the new names usually do not agree with the getCode() method.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/notice/StopLocationWithoutParentStationNotice.java (hunk 1 above)
Comment (lionel-nj) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/notice/StopLocationWithoutParentStationNotice.java:29]: @barbeau  suggested in https://github.com/MobilityData/gtfs-validator/pull/781#discussion_r585866237 we 

> put the entity name first when it makes sense for consistency, and because if we list the rules in the table alphabetically then, for example, most of the Route rules should be grouped together. 

This change would allow to regroup notices related to `stops.txt`

File: main/src/main/java/org/mobilitydata/gtfsvalidator/notice/AgencyLangInconsistencyNotice.java (hunk 2 above)
Comment (lionel-nj) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/notice/AgencyLangInconsistencyNotice.java:26]: @barbeau  suggested in https://github.com/MobilityData/gtfs-validator/pull/781#discussion_r585866237 we 

> put the entity name first when it makes sense for consistency, and because if we list the rules in the table alphabetically then, for example, most of the Route rules should be grouped together. 

This change would allow to regroup notices related to `agency.txt`

File: main/src/main/java/org/mobilitydata/gtfsvalidator/notice/AgencyTimezoneInconsistencyNotice.java (hunk 3 above)
Comment (lionel-nj) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/notice/AgencyTimezoneInconsistencyNotice.java:26]: @barbeau  suggested in https://github.com/MobilityData/gtfs-validator/pull/781#discussion_r585866237 we 

> put the entity name first when it makes sense for consistency, and because if we list the rules in the table alphabetically then, for example, most of the Route rules should be grouped together. 

This change would allow to regroup notices related to `agency.txt`

File: main/src/main/java/org/mobilitydata/gtfsvalidator/notice/StopPlatformWithoutParentStationNotice.java (hunk 6 above)
Comment (lionel-nj) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/notice/StopPlatformWithoutParentStationNotice.java:28]: @barbeau  suggested in https://github.com/MobilityData/gtfs-validator/pull/781#discussion_r585866237 we 

> put the entity name first when it makes sense for consistency, and because if we list the rules in the table alphabetically then, for example, most of the Route rules should be grouped together. 

This change would allow to regroup notices related to `stops.txt`

File: main/src/main/java/org/mobilitydata/gtfsvalidator/notice/RouteSameNameAndDescriptionNotice.java (hunk 5 above)
Comment (lionel-nj) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/notice/RouteSameNameAndDescriptionNotice.java:32]: Indeed! Will change this. thanks :)

File: main/src/main/java/org/mobilitydata/gtfsvalidator/notice/RouteSameNameAndDescriptionNotice.java (hunk 4 above)
Comment (lionel-nj) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/notice/RouteSameNameAndDescriptionNotice.java]: Removed in https://github.com/MobilityData/gtfs-validator/pull/792/commits/7296922fb39a2ce31b79a67345180d082422c57a. Thanks!

**Comment by lionel-nj** [comment]
> I see that the new names usually do not agree with the getCode() method.

`getCode()` methods implementation has been revisited to match the new class names in https://github.com/MobilityData/gtfs-validator/pull/792/commits/c0e7c010569987baf935a9c725176882753a3e6b. Thanks for your review @aababilov

**Comment by lionel-nj** [comment]
This should be re-evaluated after merging https://github.com/MobilityData/gtfs-validator/pull/822/.

**Comment by lionel-nj** [comment]
Closing as duplicates of #828
